traffic/algorithms/navigation.py

```diff
@@ -209,6 +209,59 @@ def on_runway(self, airport: Union[str, "Airport"]) -> Optional["Flight"]:
             default=None,
         )
 
+    @flight_iterator
+    def takeoff(
+        self, airport: Union[str, "Airport"]
+    ) -> Iterator["Flight"]:
+        """Iterates on all segments of trajectory matching a zone around a runway of the
+        given airport.
+        Example usage:
+        >>> 
+        """
+        
+        from shapely.geometry import Polygon
+        import traffic.core.geodesy as geo
+        from ..data import airports
+
+        # Donne les fonctions possibles sur un flight object
+        self = cast("Flight", self)
+        
+        nb_run = len(airports[airport].runways.data)
```

Comment:
What if `airports[airport]` is None, copy-paste the pattern in other similar functions.